pyk/src/pyk/cterm/cterm.py

```diff
@@ -94,8 +94,12 @@ def _check_config(config: KInner) -> None:
             raise ValueError(f'Expected cell label, found: {config}')
 
     @staticmethod
-    def _normalize_constraints(constraints: Iterable[KInner]) -> tuple[KInner, ...]:
-        constraints = sorted(normalize_constraints(constraints), key=CTerm._constraint_sort_key)
+    def _normalize_constraints(constraints: Iterable[KInner], sort_constraints: bool = True) -> tuple[KInner, ...]:
+        if sort_constraints:
+            constraints = normalize_constraints(constraints)
+            # constraints = sorted(normalize_constraints(constraints), key=CTerm._constraint_sort_key)
+        else:
+            constraints = normalize_constraints(constraints)
```

Comment:
@PetarMax Here is the fastest way to solve the disordered problem for `CSubst`

Reply:
I think there's a copy-paste error here: even if `sort_constraints=True`, the constraints won't be sorted.

Reply:
Yes, it's an issue that I don't know how to solve it. Because the following code will call `CTerm.__init__` by default and make the previous order-keeping efforts useless.

https://github.com/runtimeverification/k/blob/c8d0bee9b64f0e83a4db805575699bae040abbf1/pyk/src/pyk/kcfg/kcfg.py#L758

Reply:
I'm not familiar with the context of this change. But my point is, instead of

```python3
def _normalize_constraints(constraints: Iterable[KInner], sort_constraints: bool = True) -> tuple[KInner, ...]:
    if sort_constraints:
        constraints = normalize_constraints(constraints)
        # constraints = sorted(normalize_constraints(constraints), key=CTerm._constraint_sort_key)
    else:
        constraints = normalize_constraints(constraints)
```

you could have

```python3
def _normalize_constraints(constraints: Iterable[KInner]) -> tuple[KInner, ...]:
    constraints = normalize_constraints(constraints)
```

Reply:
Thank you! I want to refactor the function this way. But I'm not sure if there is any other applications / functions relying on the `sorted` function. Do you think it's safe to just remove the `sorted`?

Reply:
One thing I can think of where the constraints being sorted is useful is for equality checking between `CTerm`-s. I'm not sure if it's a common use case in the prover though.

Just to make sure, please check the history of this piece of code. Then we'll have a better idea of why constraints are sorted in the first place.

Reply:
I think the sorted function is introduced by Petar in [this PR](https://github.com/runtimeverification/k/pull/4281). What's your suggesions? @PetarMax